src/Controls/src/Core/Layout/GridLayout.cs

```diff
@@ -31,81 +72,101 @@ public override void Remove(IView child)
 
 		public int GetColumn(IView view)
 		{
-			return _viewInfo[view].Col;
+			return view switch
+			{
+				BindableObject bo => (int)bo.GetValue(ColumnProperty),
+				_ => _viewInfo[view].Col,
+			};
```

Comment:
What will happen when we move these layouts to Core? Do we need some sort of attached property system in Core? I would expect there to be a way to set a col, then move between grids and it will keep it.

Reply:
Or, is this never going to core and just the layout engine is there? In that case, do we care to support anything other than say BO or even the base View?

Reply:
The layout manager is in Core, along with the interface. The actual implementation of GridLayout is a Controls-only thing. And for BindableObjects, you _can_ set column/row/span info and then move it to another layout, because of how attached properties work. But that's just a weird consequence of attached properties; I wouldn't consider it an expected feature. 

There's nothing stopping a 3rd-party library from implementing MAUI Core-only objects that aren't derived from BindableObject, and there's nothing stopping the layout system from intermixing them with Controls stuff. So it seems like we ought to support that. 

For the moment, the row/column/span data has to be stored _somewhere_, so it's in a Dictionary. If we decide to add something like an attached property mechanism to Core, then we could move to using that, but that's a discussion for if and when.